service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -193,6 +202,85 @@ public void addDirectionsToCaseDetails(CaseDetails caseDetails,
+
+                    // if assigned to specific person and representatives exist -> find representative ids
+                    // if not -> return direction.
+                    if (directionIsValidForFiltering(element, representatives)) {
+                        if (element.getValue().getParentsAndRespondentsAssignee() != null) {
+                            int assigneeValue = element.getValue().getParentsAndRespondentsAssignee().ordinal();
+
+                            List<Element<Respondent>> respondents =
+                                (List<Element<Respondent>>) caseDetails.getData().get("respondents1");
+
+                            // if respondent exist -> set ids.
+                            // else return direction
+                            if (assigneeValue < respondents.size()) {
+                                ids = respondents.get(assigneeValue).getValue().getRepresentedBy();
+                            } else {
+                                return direction;
+                            }
+                        }
+
+                        if (element.getValue().getOtherPartiesAssignee() != null) {
+                            int assigneeValue = element.getValue().getOtherPartiesAssignee().ordinal();
+
+                            Others others = (Others) caseDetails.getData().get("others");
+
+                            List<Element<Other>> listOthers =
+                                defaultIfNull(others.getAdditionalOthers(), new ArrayList<>());
+
+                            listOthers.add(0, ElementUtils.element(others.getFirstOther()));
+
+                            if (assigneeValue < listOthers.size()) {
+                                ids = listOthers.get(assigneeValue).getValue().getRepresentedBy();
+                            } else {
+                                return direction;
+                            }
+                        }
+
+                    } else {
+                        // include direction in filter
+                        return direction;
+                    }
+
+                    List<Element<UUID>> finalIds = ids;
+
+                    if (ids.isEmpty()) {
+                        //
+                        return direction;
+                    }
+
+                    List<Element<Representative>> filteredRepresentatives = representatives.stream()
+                        .filter(x -> unwrapElements(finalIds).contains(UUID.fromString(x.getValue().getIdamId())))
+                        .filter(x -> x.getValue().getServingPreferences() != DIGITAL_SERVICE)
+                        .collect(toList());
+
+                    // if representative left in list, keep direction.
+                    // could be improved with a call to idam? return directions that belong to representative
+                    return !filteredRepresentatives.isEmpty();
```

Comment:
Don't like collect, maybe .findAny().isPresent() would be better?

Reply:
I will do return -> filter(find the ids that match).anyMatch(DIGITAL_SERVICE).